modules/node-shared/src/test/scala/org/tessellation/node/shared/domain/seedlist/HistoricalSnapshotOrdinalsSuite.scala

```diff
@@ -0,0 +1,260 @@
+          h.values.map(r => (r.start, r.end))
+        })
+        val expected = entries.asRight[DecoderError]
+
+        expect.same(expected, actual)
+    }
+  }
+
+  test("deserialize terminated ranges with trailing entry spaces") {
+    val gen = for {
+      a <- Gen.chooseNum(1, 5)
+      b <- Gen.listOfN(a, terminatedRangeGen)
+      c = b.foldLeft("") {
+        case (acc, (_, _, str)) =>
+          if (acc.isEmpty)
+            str
+          else
+            acc + ";" + str + "\t"
+      }
+      d = b.map { case (a, b, _) => (a, b) }
+      entries = reduce(d)
+    } yield (entries, c)
+
+    forall(gen) {
+      case (entries, str) =>
+        val actual = optionalHistoricalSnapshotOrdinalsCellDecoder(str).map(_.map { h =>
+          h.values.map(r => (r.start, r.end))
+        })
+        val expected = entries.asRight[DecoderError]
+
+        expect.same(expected, actual)
+    }
+  }
+
+  test("deserialize terminated ranges with leading entry space") {
+    val gen = for {
+      a <- Gen.chooseNum(1, 5)
+      b <- Gen.listOfN(a, terminatedRangeGen)
+      c = b.foldLeft("") {
+        case (acc, (_, _, str)) =>
+          if (acc.isEmpty)
+            str
+          else
+            acc + "; " + str
+      }
+      d = b.map { case (a, b, _) => (a, b) }
+      entries = reduce(d)
+    } yield (entries, c)
+
+    forall(gen) {
+      case (entries, str) =>
+        val actual = optionalHistoricalSnapshotOrdinalsCellDecoder(str).map(_.map { h =>
+          h.values.map(r => (r.start, r.end))
+        })
+        val expected = entries.asRight[DecoderError]
+
+        expect.same(expected, actual)
+    }
+  }
+
+  test("no overlap is returned as is") {
+    val entries: List[(NonNegLong, NonNegLong)] = List(
+      (NonNegLong.unsafeFrom(0L), NonNegLong.unsafeFrom(49L)),
+      (NonNegLong.unsafeFrom(50L), NonNegLong.unsafeFrom(59L)),
+      (NonNegLong.unsafeFrom(60L), NonNegLong.unsafeFrom(100L)),
+      (NonNegLong.unsafeFrom(100L), NonNegLong.MaxValue)
+    )
+
+    val str = "0-49;50-59;60-100;100"
+    val actual = optionalHistoricalSnapshotOrdinalsCellDecoder(str).map(_.map { h =>
+      h.values.map(r => (r.start, r.end))
+    })
+    val expected = reduce(entries).asRight[DecoderError]
+
+    expect.same(expected, actual).pure
+  }
+
+  test("unordered input does not change the output") {
+    val entries: List[(NonNegLong, NonNegLong)] = List(
+      (NonNegLong.unsafeFrom(100L), NonNegLong.MaxValue),
+      (NonNegLong.unsafeFrom(0L), NonNegLong.unsafeFrom(49L)),
+      (NonNegLong.unsafeFrom(50L), NonNegLong.unsafeFrom(59L)),
+      (NonNegLong.unsafeFrom(60L), NonNegLong.unsafeFrom(100L))
+    )
+
+    val str = "0-49;50-59;60-100;100"
+    val actual = optionalHistoricalSnapshotOrdinalsCellDecoder(str).map(_.map { h =>
+      h.values.map(r => (r.start, r.end))
+    })
+    val expected = reduce(entries).asRight[DecoderError]
+
+    expect.same(expected, actual).pure
+  }
+
+  test("overlapping ranges are merged") {
```

Comment:
please keep consistent naming. either `single, terminaned range is deserialized` and `overlapping ranges are merged` or `deserialize single, terminated range` and `merges overlapping ranges`.

Reply:
The tests have been split up into `MergeRangesSuite`, `ParseRangesSuite`, and `SnapshotOrdinalTimelineSuite`.